msc_pygeoapi/process/weather/extract_sounding_data.py

```diff
@@ -0,0 +1,640 @@
+    default=True,
+)
+@click.option(
+    '-n',
+    '--novalues_above_100mbar',
+    is_flag=True,
+    help='Set flag if you don\'t need values for pressure levels above 100mb',
+    required=False,
+)
+def extract_sounding_data_cli(
+    ctx,
+    model,
+    model_run,
+    forecast_hour,
+    lon,
+    lat,
+    convection_indices,
+    temperature_data,
+    wind_data,
+    novalues_above_100mbar,
+):
+    import rasterio
+    import time
+
+    start = time.time()
+    with rasterio.Env():
+        output = extract_sounding_data(
+            model,
+            model_run,
+            forecast_hour,
+            float(lon),
+            float(lat),
+            convection_indices,
+            temperature_data,
+            wind_data,
+            novalues_above_100mbar,
+        )
+    end = time.time()
+    print('Time elapsed array:', end - start)
+    click.echo(json.dumps(output, ensure_ascii=False))
+
+
+extract_sounding_data_execute.add_command(extract_sounding_data_cli)
+
+try:
+    from pygeoapi.process.base import BaseProcessor, ProcessorExecuteError
+
+    class ExtractSoundingDataProcessor(BaseProcessor):
+        """Extract Sounding Data Processor"""
+
+        def __init__(self, provider_def):
+            """
+            Initialize object
+            :param provider_def: provider definition
+            :returns: pygeoapi.process.weather.extract_sounding_data.ExtractSoundingDataProcessor  # noqa
+            """
+
+            BaseProcessor.__init__(self, provider_def, PROCESS_METADATA)
+
+        def execute(self, data):
+            mimetype = 'application/json'
+
+            model = data.get('model')
+            mr = data.get('model_run')
+            fh = data.get('forecast_hour')
+            lon = float(data.get('lon'))
+            lat = float(data.get('lat'))
+            conv = data.get('convection_indices')
+            temperature_data = data.get('temperature_data')
+            wind_data = data.get('wind_data')
+            noval_above_100 = data.get('novalues_above_100mbar') or False
+
+            conv, temperature_data, wind_data = [
+                arg if arg is not None else True
+                for arg in [conv, temperature_data, wind_data]
+            ]
+
+            try:
+                output = extract_sounding_data(
+                    model,
+                    mr,
+                    fh,
+                    lon,
+                    lat,
+                    conv,
+                    temperature_data,
+                    wind_data,
+                    noval_above_100,
+                )
+            except ValueError as err:
+                msg = 'Process execution error: {}'.format(err)
+                LOGGER.error(msg)
+                raise ProcessorExecuteError(msg)
+
+            dict_ = output
+
+            return mimetype, dict_
+
+        def __repr__(self):
+            return '<ExtractSoundingDataProcessor> {}'.format(self.name)
+
+
+except (ImportError, RuntimeError):
```

Comment:
Suggest to log a debug or warning

Reply:
Not sure exactly where you mean. You want a debug or warning inside the `except:` ? If so do you want me to do the same for `extract_raster.py` since that's where this line comes from?

Reply:
Yes.  +1 to update `extract_raster.py` as well.

```python
except (ImportError, RuntimeError) as err:
    LOGGER.warning('Import errors: {}'.format(err))
```